app/src/main/java/com/gpayinterns/chat/LoginActivity.java

```diff
@@ -207,11 +198,88 @@ private void setCurrentUser()
         mEditor.putString("currentUser", currentUser).apply();
     }
 
-    private boolean authenticate(String email, String password)
+
+    private void authenticateFromServer() throws JSONException
     {
-        //TODO authenticate from backend server
-        //currentUser = userID received from the server
-        return true;
+        String userName = usernameEditText.getText().toString();
+        String password = passwordEditText.getText().toString();
+
+        String URL = "https://gcp-chat-service.an.r.appspot.com/login";
+
+
```

Comment:
Why 2 blank lines?
1 is enough.

Reply:
This change has been done on "Android: Encapsulated Server URL Details + view time for messages"